internal/command/junit/junit.go

```diff
@@ -206,31 +218,41 @@ func junitXMLTestReport(suite *moduletest.Suite, sources map[string][]byte) ([]b
 					// why the test was skipped?
 				}
 			case moduletest.Fail:
+				var failedAssertion tfdiags.Diagnostic
+				for key, d := range diagsMap {
+					// Find the diag resulting from a failed assertion
+					if d.Description().Summary == failedTestSummary {
+						failedAssertion = d
+						delete(diagsMap, key)
+						break
```

Comment:
Is there a reason we always report just the first failed assertion rather than all of them? Or do we assume that there will always be at most 1 diagnostic representing a failed assertion?

Reply:
You're right - if there are multiple failing assertions this PR will currently use only the first one and dump the remaining ones in `<system-err>`. Though the problem is that there can only be one `<failure>` element and if there are multiple assertion failures it's not clear what the `message` attribute value should then be.

It could be that `message` contains the assertion's error message if there's a single failure, and a summary if there are multiple? E.g:

- 1 assertion failure: `"1 assertion failure: <error message here>"`
- 3 assertion failures: `"3 assertion failures, including: <error message here>"`

What do you think?

Reply:
I've addressed this in [b6798ae](https://github.com/hashicorp/terraform/pull/36316/commits/b6798ae702022eefdf524c914bedc1dfaa78a58c)

That change makes sure all test failure diags are used in "failure" elements. The `message` reports how many assertions failed out of X and includes the first assertion failure message.